src/screens/OtherProfile/elements/BioAndChat.js

```diff
@@ -0,0 +1,190 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {Pressable, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
+
+import TextAreaChat from '../../../components/TextAreaChat';
+import ToggleSwitch from '../../../components/ToggleSwitch';
+import useSaveAnonChatHook from '../../../database/hooks/useSaveAnonChatHook';
+import {ANON_PM, SIGNED} from '../../../hooks/core/constant';
+import {colors} from '../../../utils/colors';
+import {fonts} from '../../../utils/fonts';
+import {sendAnonymousDMOtherProfile, sendSignedDMOtherProfile} from '../../../service/chat';
+import {trimString} from '../../../utils/string/TrimString';
+
+const CHANNEL_BLOCKED = 'Channel is blocked';
+
+const BioAndChat = (props) => {
+  const {
+    isAnonimity,
+    bio,
+    openBio,
+    dataMain,
+    isSignedMessageEnabled,
+    showSignedMessageDisableToast,
+    loadingGenerateAnon,
+    avatarUrl,
+    anonProfile,
+    username,
+    toggleSwitch,
+    isAnonimityEnabled
+  } = props;
+  const {saveChatFromOtherProfile, savePendingChatFromOtherProfile} = useSaveAnonChatHook();
+  const [dmChat, setDmChat] = React.useState('');
+  const [loadingSendDM, setLoadingSendDM] = React.useState(false);
+
+  React.useEffect(() => {
+    setDmChat('');
+  }, [isSignedMessageEnabled]);
+
+  const sendSignedDM = async () => {
+    try {
+      setLoadingSendDM(true);
+      const signedMParams = {
+        user_id: dataMain.user_id,
+        message: dmChat
+      };
+      const response = await sendSignedDMOtherProfile(signedMParams);
+      const newResponse = {...response, members: response?.message?.members};
+      await saveChatFromOtherProfile(newResponse, 'sent', true, SIGNED);
+      setDmChat('');
+    } catch (error) {
+      if (__DEV__) {
+        console.log(error, 'error');
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

-      if (__DEV__) {
-        console.log(error, 'error');
-      }
+      // Show an error message to the user or handle the error appropriately



<!-- This is an auto-generated comment by OSS CodeRabbit -->